test/hotspot/jtreg/runtime/cds/SharedStrings.java

```diff
@@ -42,27 +42,28 @@ public static void main(String[] args) throws Exception {
         // This also serves as a reference on how to use this feature,
         // hence the command lines are spelled out instead of using the
         // test utils methods.
-        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
-            "-XX:+UnlockDiagnosticVMOptions",
-            "-XX:SharedArchiveFile=./SharedStrings.jsa",
-            "-Xlog:cds,cds+hashtables",
-            // Needed for bootclasspath match, for CDS to work with WhiteBox API
-            "-Xbootclasspath/a:" + ClassFileInstaller.getJarPath("whitebox.jar"),
-            "-Xshare:dump");
-
-        OutputAnalyzer out = CDSTestUtils.executeAndLog(pb, "dump");
+        CDSOptions opts = (new CDSOptions())
+            .addPrefix("-XX:+UnlockDiagnosticVMOptions",
+                       "-Xlog:cds,cds+hashtables",
+                       // Needed for bootclasspath match, for CDS to work with WhiteBox API
+                       "-Xbootclasspath/a:" + ClassFileInstaller.getJarPath("whitebox.jar"))
+            .setArchiveName("./SharedStrings.jsa");
+        OutputAnalyzer out = CDSTestUtils.createArchive(opts);
         CDSTestUtils.checkDump(out, "Shared string table stats");
 
-
-        pb = ProcessTools.createJavaProcessBuilder(
-                "-XX:+UnlockDiagnosticVMOptions",
-                "-XX:SharedArchiveFile=./SharedStrings.jsa",
-                // needed for access to white box test API
-                "-Xbootclasspath/a:" + ClassFileInstaller.getJarPath("whitebox.jar"),
-                "-XX:+UnlockDiagnosticVMOptions", "-XX:+WhiteBoxAPI",
-                "-Xshare:on", "-showversion", "SharedStringsWb");
-
-        out = CDSTestUtils.executeAndLog(pb, "exec");
-        CDSTestUtils.checkExec(out);
+        opts = (new CDSOptions())
+            .setUseVersion(false)
+            // This test is run with the -Xshare:off vmoption prepended in tier3.
+            // Set CDSOptions.xShareMode to "off" so that the result checking
+            // would not always expect the word "sharing" in the output.
+            .setXShareMode("off")
```

Comment:
According to CDSTestUtils.runWithArchive:
      for (String p : opts.prefix) cmd.add(p);

        cmd.add("-Xshare:" + opts.xShareMode);
The -Xshare::off will be the last one for share mode setting, so this test will be always with be with shared off?
Unless the suffix include share setting to change that.

Reply:
I took a look at some old test log of this test and noticed although the following was set:
`-Dtest.java.opts='-Xshare:off -showversion'`

The options were never included in the command line because the `executeAndLog` was used.
There's also the following comment in SharedStrings.java:        
```
        // Note: This is a basic sanity test for Shared Strings feature.
        // This also serves as a reference on how to use this feature,
        // hence the command lines are spelled out instead of using the
        // test utils methods.
```

I'm proposing revert the change of this file.